### PR TITLE
chore: Check for Rook pods to rollout completely and update Rook to 1.10.11

### PR DESCRIPTION
**Pull request by rrpolanco**
#### What this PR does / why we need it:

Use Rook util functions introduced in https://github.com/replicatedhq/kURL/pull/4063 to wait for Rook to rollout after installation/upgrade.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
Adds [Rook add-on](https://kurl.sh/docs/add-ons/rook) version 1.10.11.
```

### CODE REVIEW

**Review by camilamacedo86 — COMMENTED** [review]
@rrpolanco,

it is failing with:

```
2023-02-14 23:07:36+00:00 configmap/kurl-current-config patched
2023-02-14 23:07:36+00:00 error: failed to create configmap: ConfigMap "kurl_cluster_uuid" is invalid: metadata.name: Invalid value: "kurl_cluster_uuid": a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')
2023-02-14 23:07:36+00:00 An error occurred on line 
+ KURL_EXIT_STATUS=1
+ export KUBECONFIG=/etc/kubernetes/admin.conf
+ KUBECONFIG=/etc/kubernetes/admin.conf
+ export PATH=/sbin:/bin:/usr/sbin:/usr/bin:/usr/local/bin
+ PATH=/sbin:/bin:/usr/sbin:/usr/bin:/usr/local/bin
+ '[' 1 -eq 0 ']'
+ echo ''
+ echo 'failed kurl install with exit status 1'
+ collect_debug_info_after_kurl
+ '[' 1 -ne 0 ']'
+ echo 'kubelet status'
```

File: scripts/Manifest [hunk 1]
```diff
@@ -11,5 +11,25 @@ KURL_BIN_UTILS_FILE=
...
+  minio:
+    version: 2023-01-31T02-24-19Z
+  kotsadm:
+    version: 1.94.0
+  ekco:
+    version: 0.26.3
+  containerd:
+    version: 1.6.16
+  sonobuoy:
+    version: 0.56.15"
```
[the remark this answers is not in the document]
Comment (camilamacedo86) [reply scripts/Manifest:35]: +1 good catcher

[the remark this answers is not in the document]
Comment (rrpolanco) [reply scripts/Manifest:35]: Yup, good catch. I'll update this.

File: addons/rook/template/base/install.sh [hunk 2]
```diff
@@ -117,6 +117,10 @@ function rook() {
 
     # migrate from Longhorn storage if applicable
     rook_maybe_migrate_from_longhorn
+
+    # wait for all pods in the rook-ceph namespace to rollout
+    log "Awaiting Rook rollout in rook-ceph namespace"
+    rook_maybe_wait_for_rollout
```
Comment (emosbaugh) [inline addons/rook/template/base/install.sh:123]: should we add this to the rook_already_applied function as well?

**Review by emosbaugh — CHANGES_REQUESTED** [review]

File: addons/rook/template/base/install.sh [hunk 3]
```diff
@@ -1015,3 +1019,16 @@ function rook_prompt_migrate_from_longhorn() {
         bail "Not migrating"
     fi
 }
+
+# wait for Rook deployment pods to be running/completed
+function rook_maybe_wait_for_rollout() {
+    log "Awaiting Rook Operator to be Ready"
+    if ! spinner_until 120 rook_operator_ready; then
```
Comment (camilamacedo86) [inline addons/rook/template/base/install.sh:1026]: Could we increase the timeouts to something like 5 min ?

**Review by camilamacedo86 — DISMISSED** [review]
I think we just need increase the timeout for like 5 min.
Otherwise, it shows OK for me.

/LGTM

File: addons/rook/template/base/install.sh (hunk 2 above)
Comment (rrpolanco) [reply addons/rook/template/base/install.sh:123]: Added: https://github.com/replicatedhq/kURL/pull/4101/files#diff-5041c635371838ae5901774a8bcf202a1543309d1462226eb36278b4a07da8f9R134

**Comment by rrpolanco** [comment]
> Testgrid Run(s) Executing @ https://testgrid.kurl.sh/run/pr-4101-590e780-rook-1.10.11-k8s-docker-2023-02-21T19:21:58Z



There are a couple of runs for `Upgrade from 1.8.10` which are failing. The ubuntu 18.04 run fails in the post-upgrade script:
```
+ kubectl apply -f -
2023-02-21 20:12:18+00:00 persistentvolumeclaim/cephfs-pvc created
2023-02-21 20:12:18+00:00 persistentvolumeclaim/default-pvc created
+ sleep 10
+ kubectl -n default get pvc default-pvc
+ grep Bound
+ local exit_status=1
+ '[' 1 -ne 0 ']'
+ report_failure post_upgrade_script
+ local failure=post_upgrade_script
```

**Comment by camilamacedo86** [comment]
HI @rrpolanco,

Prior the latest change it shows OK, here: https://github.com/replicatedhq/kURL/pull/4101#issuecomment-1439055581 see:
- The error faced with `Upgrade from 1.8.10` is just because we could not check the PVC bound in 10 seconds
- The errors faced with `Rook migrate from Longhorn` shows not been because we are adding a new version the

But now, that is seems broken after the latest changes : https://testgrid.kurl.sh/run/pr-4101-d8401ff-rook-1.10.11-k8s-docker-2023-02-21T20:39:16Z

**Review by emosbaugh — APPROVED** [review]
This looks good to me although based on the testgrid results I think we still have some work to do to wait long enough for rook to be ready to serve requests for pvs

**Review by emosbaugh — APPROVED** [review]

**Pull request merged**